Pressingly/lago-api
# PINET-160: set up Amazon Verified Permission

## Context

Set up Amazon Verified Permission (AVP) as the first part of story PINET-160 https://thepressingly.atlassian.net/browse/PINET-160

## Description

To avoid reimplement policy store, we've decided to go with AVP for https://docs.google.com/document/d/1d7Q8AUPzRTPKge6DMwOXWL1JphSeeT21opeAJFxQqks/edit, therefore this PR is to set up AWS verification permission

## How to test

- Firstly, add these values to `.env` file
```bash
AWS_ACCESS_KEY_ID="YOUR AWS ACCESS KEY"
AWS_SECRET_ACCESS_KEY="YOUR AWS SECRET ACCESS KEY"
```
For more information about how to acquire these values, pls visit this doc https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html#Using_CreateAccessKey
- Run this repo at your local with this command `./scripts/start.dev.sh`
- Get you policy store id from your aws console
<img width="1351" alt="image" src="https://github.com/Pressingly/lago-api/assets/33661599/3bbf820f-fb4b-4300-b672-f6c9f9ffd6db">
and pass it to `policy_store_id` , which looks like this
<img width="790" alt="image" src="https://github.com/Pressingly/lago-api/assets/33661599/ec9c46ee-7215-4340-b910-145f4653d2fc">

- Go to the browser at this endpoint `http://localhost:3000/entitlement/authorization`, you should see something like the screenshot below
<img width="1422" alt="image" src="https://github.com/Pressingly/lago-api/assets/33661599/b61776a6-3e18-4d6d-bc3d-91514e8c6757">
